q2_diversity/tests/test_alpha.py

```diff
@@ -19,189 +19,121 @@
+                                                       three_feature_tree)
 
-    def test_alpha_phylogenetic_old(self):
         t = biom.Table(np.array([[0, 1, 3], [1, 1, 2]]),
                        ['O1', 'O2'],
                        ['S1', 'S2', 'S3'])
+        self.t = Artifact.import_data('FeatureTable[Frequency]', t)
         tree = skbio.TreeNode.read(io.StringIO(
             '((O1:0.25, O2:0.50):0.25, O3:0.75)root;'))
-        actual = alpha_phylogenetic_old(
-            table=t, phylogeny=tree, metric='faith_pd')
-        # expected computed with skbio.diversity.alpha_diversity
-        expected = pd.Series({'S1': 0.75, 'S2': 1.0, 'S3': 1.0},
-                             name='faith_pd')
+        self.tree = Artifact.import_data('Phylogeny[Rooted]', tree)
+
+    def test_alpha(self):
+        actual = self.alpha(table=self.t, metric='observed_features')
+        actual = actual[0].view(pd.Series)
+        # expected computed by hand
+        expected = pd.Series({'S1': 1, 'S2': 2, 'S3': 2},
+                             name='observed_features')
         pdt.assert_series_equal(actual, expected)
 
-    def test_alpha_phylogenetic_old_non_phylo_metric(self):
-        t = biom.Table(np.array([[0, 1, 3], [1, 1, 2]]),
-                       ['O1', 'O2'],
-                       ['S1', 'S2', 'S3'])
-        tree = skbio.TreeNode.read(io.StringIO(
-            '((O1:0.25, O2:0.50):0.25, O3:0.75)root;'))
-        with self.assertRaisesRegex(ValueError, 'Unknown phylogenetic metric'):
-            alpha_phylogenetic_old(
-                table=t, phylogeny=tree, metric='observed_otus')
+    def test_alpha_with_passthrough_metric(self):
+        actual = self.alpha(table=self.t, metric='singles')
+        actual = actual[0].view(pd.Series)
+        # expected computed by hand
+        expected = pd.Series({'S1': 1, 'S2': 2, 'S3': 0},
+                             name='singles')
+        pdt.assert_series_equal(actual, expected)
 
-    def test_alpha_phylogenetic_old_unknown_metric(self):
-        t = biom.Table(np.array([[0, 1, 3], [1, 1, 2]]),
-                       ['O1', 'O2'],
-                       ['S1', 'S2', 'S3'])
-        tree = skbio.TreeNode.read(io.StringIO(
-            '((O1:0.25, O2:0.50):0.25, O3:0.75)root;'))
-        with self.assertRaisesRegex(ValueError, 'Unknown phylogenetic metric'):
-            alpha_phylogenetic_old(
-                table=t, phylogeny=tree, metric='not-a-metric')
+    def test_alpha_phylo_metric(self):
+        with self.assertRaisesRegex(TypeError, 'faith_pd.*incompatible'):
+            self.alpha(table=self.t, metric='faith_pd')
+        # Check vanilla python function as well
+        with qiime2.sdk.Context() as scope:
+            with self.assertRaisesRegex(ValueError,
+                                        'Unknown metr.*not-a-metric'):
+                alpha(ctx=scope.ctx, table=self.t, metric='not-a-metric')
```

Comment:
The second half of this test is unnecessary, and I think it should be removed:

```suggestion
    def test_alpha_phylo_metric(self):
        with self.assertRaisesRegex(TypeError, 'faith_pd.*incompatible'):
            self.alpha(table=self.t, metric='faith_pd')
```

Reply:
Please remove any other similar "vanilla python" checks, as well.

Reply:
These were added as much to fix test coverage (#255), as they were to support vanilla python imports. Now that the 'invalid metric' guards have been removed from the alpha and beta dispatch functions, these have been removed without negatively impacting test coverage.